hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/TransactionChecker.java

```diff
@@ -362,18 +372,86 @@ private void checkTransactionBody(@NonNull final TransactionBody txBody, HederaF
+        // (since payer-based limits are applied in checkTransactionSizeLimitBasedOnPayer)
+        // Only reject extremely oversized transactions that exceed governance limits
+        else {
+            exceedsLimit = txSize > governanceTransactionsConfig().maxTxnSize();
+        }
+
+        if (exceedsLimit) throw new PreCheckException(TRANSACTION_OVERSIZE);
+    }
+
+    /**
+     * Validates transaction size limits based on the payer account's privileges.
+     * @param txInfo the {@link TransactionInfo} to check
+     * @param payerAccountId the {@link AccountID} of the transaction payer
+     */
+    public void checkTransactionSizeLimitBasedOnPayer(
+            @NonNull final TransactionInfo txInfo, @NonNull final AccountID payerAccountId) throws PreCheckException {
+        boolean exceedsLimit;
+        final int txSize = txInfo.signedTx().protobufSize();
+        // Check if the payer is privileged (treasury or systemAdmin)
+        final boolean isPrivilegedPayer =
+                accountsConfig().governanceTransactions().contains(payerAccountId.accountNumOrElse(0L));
+
+        if (isPrivilegedPayer) {
+            exceedsLimit = txSize > governanceTransactionsConfig().maxTxnSize();
+        } else {
+            // Non-privileged payers follow jumbo logic when the jumbo feature is enabled
+            if (jumboTransactionsConfig().isEnabled()) {
+                exceedsLimit = checkJumboRequirements(txInfo);
+            } else {
+                // Only governance enabled, non-privileged payers are limited to standard size
+                exceedsLimit = txSize > hederaConfig().transactionMaxBytes()
+                        && !NON_JUMBO_TRANSACTIONS_BIGGER_THAN_6_KB.contains(txInfo.functionality());
+            }
+        }
 
-        if (jumboTxnEnabled
-                && txInfo.signedTx().protobufSize() > hederaConfig.transactionMaxBytes()
-                && !allowedJumboHederaFunctionalities.contains(fromPbj(txInfo.functionality()))
-                && !NON_JUMBO_TRANSACTIONS_BIGGER_THAN_6_KB.contains(txInfo.functionality())) {
+        if (exceedsLimit) {
+            if (!isPrivilegedPayer) {
+                nonPrivilegedOversizedTransactionsCounter.increment();
+            }
             throw new PreCheckException(TRANSACTION_OVERSIZE);
         }
     }
 
+    /**
+     * Validates can be considered a jumbo transaction.
+     * @param txInfo the {@link TransactionInfo} to check
+     * @return whether the transaction is considered jumbo
+     */
+    private boolean checkJumboRequirements(@NonNull final TransactionInfo txInfo) {
+        final int txSize = txInfo.signedTx().protobufSize();
+        final HederaFunctionality functionality = txInfo.functionality();
+        final var allowedJumboHederaFunctionalities = jumboTransactionsConfig().allowedHederaFunctionalities();
+
+        return !allowedJumboHederaFunctionalities.contains(fromPbj(txInfo.functionality()))
```

Comment:
could `functionality` be used here instead of `txInfo.functionality()`?

also a nit but IMO this boolean value is difficult to read and validate.